tests/utils/test_safe_crash.py

```diff
@@ -114,6 +114,33 @@ def match_once():
     pytest.fail(message)
 
 
+@pytest.mark.xfail(strict=False, reason="Compiled regex still crashes under PYTHON_GIL=0")
+def test_compiled_regex_thread_safety_crashes_under_gil0():
+    script = _regex_thread_script(
+        imports="import regex",
+        setup_code="""
+        compiled_pattern = regex.compile(pattern_text)
+
+        def match_once():
+            return compiled_pattern.match(text_to_match)
+        """,
+    )
+
+    result, message = _run_regex_thread_script("compiled regex", script)
+
+    if result.returncode == 0:
+        pytest.fail("compiled regex unexpectedly behaved thread-safely\n" + message)
+
+    if result.returncode == -11:
+        message += "\nProcess terminated with SIGSEGV (Segmentation fault)."
+
+    if message:
+        sys.stderr.write(message + "\n")
+        sys.stderr.flush()
+
+    pytest.fail(message)
+
+
```

Comment:
Add new test to show `regex` is crashing with GIL=0 even when the pattern is not going through the caching mechanism which I thought was the cause of the previous crashes. This is strange and unexpected. It appears `regex` lib is fully unsafe as it maybe reusing the same runtime stack/buffers for same pattern execution.